docs/src/main/asciidoc/cyclonedx.adoc

```diff
@@ -0,0 +1,192 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
+////
+[id="cyclonedx"]
+= Generating CycloneDX BOMs
+include::_attributes.adoc[]
+:categories: tooling
+:summary: This guide explains how to generate SBOMs for Quarkus applications in the CycloneDX format.
+:topics: sbom
+:extensions: io.quarkus:quarkus-cyclonedx
+
+An SBOM (Software Bill of Material) is a manifest that describes what a given software distribution consists of in terms of components. In addition to that, it may include a lot more information such as relationships between those components, licenses, provenance, etc.
+SBOMs would typically be used by software security and software supply chain risk management tools to perform vulnerability and compliance related analysis.
+
+This guide describes Quarkus SBOM generation capabilities following https://cyclonedx.org/[CycloneDX] specification.
+
+== Why Quarkus-specific tooling?
+
+While Quarkus integrates with build tools such as https://maven.apache.org/[Maven] and https://gradle.org/[Gradle], it could itself be categorized as a build tool with its own component and dependency model, build steps and build outcomes. One of the essential component types of a Quarkus application is a Quarkus extension, which consists of a runtime and a build time artifacts, and their dependencies.
+
+To properly resolve Quarkus extension and other application dependencies Quarkus uses its own dependency resolver, which is implemented on top of the dependency resolver provided by the underlying build tool: Maven or Gradle.
+
+As a consequence, in case of Maven, for example, the results of `dependency:tree` will not include all the dependencies Quarkus will use to build an application. A similar issue will affect other dependency analysis tools that assume a project adheres to the standard Maven dependency model: they will not be able to capture the effective Quarkus application dependency graph. That, unfortunately, includes the https://github.com/CycloneDX/cyclonedx-maven-plugin[CycloneDX Maven plugin] implementation.
```

Comment:
```suggestion
As a consequence, in case of Maven, for example, the results of `dependency:tree` will not include all the dependencies Quarkus will use to build an application. A similar issue will affect other dependency analysis tools that assume a project adheres to the standard Maven dependency model: they will not be able to capture the effective Quarkus application dependency graph. Unfortunately, that includes the implementation of the https://github.com/CycloneDX/cyclonedx-maven-plugin[CycloneDX Maven plugin].
```